### ISSUE TITLE
Year

### ISSUE THREAD

**Issue by pablojoii** [iamvivekkaushik/DatePickerTimelineFlutter#92]
Could you add some params (boolean option, fontstyle only) for Year?

I mean, I'm in 2024, and want to scroll back to 2012 - when there were no apps, hahaha - and I have to count the Januaries? :+1: 

Get the point? And the information is there already, just show it. 

yearOnLeft: bool
yearOnRight: bool
yearOnCenter: bool
yearStyle: ...

:-)

**Comment by iamvivekkaushik** [comment]
Fair point — counting Januaries back to 2012 is not a UX anyone deserves. :-)

Honest status: the day timeline still doesn't have a year label, and I'd
rather not bolt on `yearOnLeft/Right/Center` params without doing it
properly — what I want is a sticky year header that updates as you scroll.
Keeping this open to track that.
Two things that help meanwhile:
1. Since **v1.3.0**, `showPastDates: true` renders past dates and opens
   centered on the selection, so scrolling back years is first-class.
2. **v1.4.0** adds `granularity: DateGranularity.week / .month` — those
   coarser tiles display the year on every tile (`AUG / 22–28 / 2026`),
   which covers the multi-year browsing case this issue describes.
For a year label on the day timeline today, render your own header driven by
`onDateChange`:
```dart
String _year = DateFormat('yyyy').format(DateTime.now());
Column(children: [
  Text(_year, style: Theme.of(context).textTheme.titleMedium),
  DatePicker(
    DateTime.now(),
    showPastDates: true,
    onDateChange: (date) =>
        setState(() => _year = DateFormat('yyyy').format(date)),
  ),
]),
